ingenialink/eth/net.c

```diff
@@ -328,11 +337,46 @@ static void il_eth_net_close_socket(il_net_t *net) {
     il_eth_net_t *this = to_eth_net(net);
 
     int r = 0;
-    r = closesocket(this->server);
-    WSACleanup();
-    return r;
+    #ifdef _WIN32
+        r = shutdown(this->server, SD_BOTH);
+        if (r == 0) { r = closesocket(this->server); }
+    #else
+        r = shutdown(this->server, SHUT_RDWR);
+        if (r == 0) { r = close(this->server); }
+    #endif
+
+    return r;	
+}
+
+static int il_eth_net_init_socket(void)
+{
+    #ifdef _WIN32
+        WSADATA wsa_data;
+        return WSAStartup(0x202, &wsa_data);
```

Comment:
What is 0x202?

Reply:
It's the highest version of Windows Sockets support that the caller can use. [Source](http://www.cs.rpi.edu/academics/courses/fall09/os/win32/WSAstartup.html)